C7/Game.cs

```diff
@@ -580,107 +443,159 @@ public override void _UnhandledInput(InputEvent @event)
-	///the keyboard shortcuts should wind up here.
-	///Eventually, we should quite possibly put this somewhere other than Game.cs, or at
-	///least the logic should be somewhere else.  I want to see how it looks with a couple
-	///more things going on before figuring out what the 'right' thing is, though.
-	private void UnitButtonPressed(string buttonName)
-	{
-		// this will detoggle goTo when clicking unit buttons
-		// other than goTo
-		setGoToMode(buttonName == "goTo");
-
-		log.Verbose("The " + buttonName + " button was pressed");
-		switch (buttonName) {
-		case "hold":
-			new MsgSkipUnitTurn(CurrentlySelectedUnit.guid).send();
-			break;
+		if (this.HasCurrentlySelectedUnit()) {
+			// TODO: replace bool with an invalid TileDirection enum
+			TileDirection dir = TileDirection.NORTH;
+			bool moveUnit = true;
+			if (Input.IsActionJustPressed("move_unit_southwest")) {
+				dir = TileDirection.SOUTHWEST;
+			} else if (Input.IsActionJustPressed("move_unit_south")) {
+				dir = TileDirection.SOUTH;
+			} else if (Input.IsActionJustPressed("move_unit_southeast")) {
+				dir = TileDirection.SOUTHEAST;
+			} else if (Input.IsActionJustPressed("move_unit_west")) {
+				dir = TileDirection.WEST;
+			} else if (Input.IsActionJustPressed("move_unit_east")) {
+				dir = TileDirection.EAST;
+			} else if (Input.IsActionJustPressed("move_unit_northwest")) {
+				dir = TileDirection.NORTHWEST;
+			} else if (Input.IsActionJustPressed("move_unit_north")) {
+				dir = TileDirection.NORTH;
+			} else if (Input.IsActionJustPressed("move_unit_northeast")) {
+				dir = TileDirection.NORTHEAST;
+			} else {
+				moveUnit = false;
+			}
+			if (moveUnit) {
+				new MsgMoveUnit(CurrentlySelectedUnit.guid, dir).send();
+				setSelectedUnit(CurrentlySelectedUnit); //also triggers updating the lower-left info box
+			}
+		}
 
-		case "fortify":
-			new MsgSetFortification(CurrentlySelectedUnit.guid, true).send();
-			break;
+		if (Input.IsActionJustPressed("toggle_grid")) {
+			this.mapView.gridLayer.visible = !this.mapView.gridLayer.visible;
+		}
+
+		if (Input.IsActionJustPressed("escape") && !this.inUnitGoToMode) {
+			log.Debug("Got request for escape/quit");
+			PopupOverlay popupOverlay = GetNode<PopupOverlay>(PopupOverlay.NodePath);
+			popupOverlay.ShowPopup(new EscapeQuitPopup(), PopupOverlay.PopupCategory.Info);
+		}
+
+		if (Input.IsActionJustPressed("toggle_zoom")) {
+			if (mapView.cameraZoom != 1) {
+				mapView.setCameraZoomFromMiddle(1.0f);
+				VSlider slider = GetNode<VSlider>("CanvasLayer/SlideOutBar/VBoxContainer/Zoom");
+				slider.Value = 1.0f;
+			} else {
+				mapView.setCameraZoomFromMiddle(0.5f);
+				VSlider slider = GetNode<VSlider>("CanvasLayer/SlideOutBar/VBoxContainer/Zoom");
+				slider.Value = 0.5f;
+			}
+		}
+
+		if (Input.IsActionJustPressed("toggle_animations")) {
+			SetAnimationsEnabled(false);
+		} else if (Input.IsActionJustReleased("toggle_animations")) {
```

Comment:
Is this part now redundant with line 428?

Reply:
it is, thanks for catching. I removed the else if condition on 428